src/client/opamClient.ml

```diff
@@ -850,14 +850,18 @@ let windows_checks ?cygwin_setup ?git_location config =
            OpamFilename.(Dir.to_string (dirname_dir (dirname_dir cygbin))));
     config
   in
+  let packages =
+    match OpamSystem.resolve_command "git" with
+    | None -> OpamInitDefaults.required_packages_for_cygwin
+    | Some _ ->
+      List.filter (fun c -> not OpamSysPkg.(equal (of_string "git") c))
+        OpamInitDefaults.required_packages_for_cygwin
+  in
   let install_cygwin_tools () =
-    let packages =
-      match OpamSystem.resolve_command "git" with
-      | None -> OpamInitDefaults.required_packages_for_cygwin
-      | Some _ ->
-        List.filter (fun c -> not OpamSysPkg.(equal (of_string "git") c))
-          OpamInitDefaults.required_packages_for_cygwin
-    in
+    OpamSysInteract.Cygwin.install ~packages
+  in
+  let install_cygwin_tools_with_pkgs ~pkgs =
```

Comment:
why not simply merge this new function with the existing `install_cygwin_tools` function?

Reply:
Ok, that's fair.